### PR TITLE
[Refactoring] Replace 'for loop' with 'enhanced for'

### PR DESCRIPTION
**Pull request by Shadow-Devil**
## Purpose
This replaces for loops which iterate over collections or arrays and can be replaced with an enhanced for loop.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by Shadow-Devil** [comment]
Could someone please have a look at this PR? Thank you ❤️

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BResourceMethodType.java [hunk 1]
```diff
@@ -53,9 +53,9 @@ public String toString() {
         }
         StringJoiner sj = new StringJoiner(",", "resource function " + accessor + " " + rp.toString() +
                 "(", ") returns (" + type.retType + ")");
-        for (int i = 0; i < parameters.length; i++) {
-            Type type = parameters[i].type;
-            sj.add(type.getName() + " " + parameters[i].name);
+        for (var parameter : parameters) {
```
Comment (MaryamZi) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BResourceMethodType.java:56]: Shall we avoid using `var` (to help with readability).

Comment (Shadow-Devil) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BResourceMethodType.java]: Fixed in [1663d93](https://github.com/ballerina-platform/ballerina-lang/pull/42905/commits/1663d93c8796e217939ab8765f7911a6f4372592)

**Review by MaryamZi — APPROVED** [review]

**Review by azinneera — APPROVED** [review]
Testerina and CLI changes LGTM.

**Review by ravinperera00 — APPROVED** [review]
LGTM

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**